packages/blade/src/components/Carousel/__tests__/Carousel.test.stories.tsx

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import type { StoryFn } from '@storybook/react';
+import { within, userEvent } from '@storybook/testing-library';
+import { expect, jest } from '@storybook/jest';
+import type { Mock } from 'jest-mock';
+import type { CarouselProps } from '../';
+import { Carousel as CarouselComponent } from '../';
+import { CarouselExample } from '../Carousel.stories';
+import { Box } from '~components/Box';
+
+const sleep = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+
+let onChange: Mock<unknown, unknown[]> | null = null;
+
+const BasicCarousel = (props: CarouselProps): React.ReactElement => (
+  <Box margin="auto" width={{ base: '100%', m: '100%' }} padding="spacing.4">
+    <CarouselExample {...props} />
+  </Box>
+);
+
+export const TestCarouselOnChange: StoryFn<typeof CarouselComponent> = (
+  props,
+): React.ReactElement => {
+  onChange = jest.fn();
+  return <BasicCarousel {...props} onChange={onChange} />;
+};
+
+TestCarouselOnChange.play = async ({ canvasElement }) => {
+  const { getByRole, getByText } = within(canvasElement);
+  const nextButton = getByRole('button', { name: 'Next Slide' });
+  const previousButton = getByRole('button', { name: 'Previous Slide' });
+  await userEvent.click(nextButton);
+  await sleep(1000);
+  await expect(onChange).toBeCalledWith(1);
+  await userEvent.click(previousButton);
+  await sleep(1000);
+  await expect(onChange).toBeCalledWith(0);
+  getByText(/Single Flow To Collect And Disburse Payments/)?.scrollIntoView({ behavior: 'smooth' });
+  await sleep(1000);
+  await expect(onChange).toBeCalledWith(3);
```

Comment:
Actually we should also do this, after all the toBeCalledWith are done we should do another check for. 

onChange.toBeCalledTimes(3); 

This will ensure that onChange hasn't been called twice or more than it needs to be. 

This we can add in all the tests.